Class check update

andrasguseo opened the pull request:
Fixed class check for Settings_Helper in order to make it work next to other extensions

Commit:
Added class check

Added a class check to Settings_Helper otherwise it would break when other extensions with SH are also used.
Not sure the first check is needed at the top of the file.

Commit:
Updated version number and readme

Commit:
Removed superfluous class check

Commit:
Removed git files

bordoni reviewed (dismissed):
`.idea` folder needs to be ignored, keep the `.gitignore` file there, no need to remove it.

Commit:
Revert "Removed git files"

This reverts commit 4bfc8f3233e77c42bcac4257b0e946dc4a64bfb1.

Commit:
Gitignore back

andrasguseo commented:
> 
> 
> `.idea` folder needs to be ignored, keep the `.gitignore` file there, no need to remove it.

Thanks for being patient with me. :) These git files are a bit confusing for me.

cliffordp commented on .gitignore:
```diff
@@ -214,3 +214,4 @@ pip-log.txt
 
 #Mr Developer
```
Please delete this entire `.gitignore` file, as our other extensions don't have one, and this repo's has a big one. Instead, add these "should be included from all my projects everywhere" type of things to your own personal computer's *global* gitignore. Instructions are here: https://help.github.com/en/github/using-git/ignoring-files

cliffordp commented on index.php:
```diff
@@ -41,7 +41,9 @@ public function construct() {
 	 * Adds settings options
 	 */
 	public function add_settings() {
-		require_once dirname( __FILE__ ) . '/src/Tribe/Settings_Helper.php';
```
Alternatively, you could _namespace_ this plugin, like the extension template does: https://github.com/mt-support/tribe-ext-extension-template

cliffordp commented on readme.txt:
```diff
@@ -35,6 +35,10 @@ We're always interested in your feedback and our [premium forums](https://theeve
 
+= 1.0.1 2020-02-14 =
+
+* Fixed class check for Settings_Helper in order to make it work next to other extensions.
```
```suggestion
* Fix - Add check for `Settings_Helper` class already loaded in order to not conflict with other extensions.
```

cliffordp commented on src/Tribe/Settings_Helper.php:
```diff
@@ -5,170 +5,168 @@
 	die( '-1' );
 }
 
-if ( class_exists( 'Tribe__Settings_Helper' ) ) {
```
I didn't review this file, assuming you just wrapped in a `class_exists()` check without other changes... but the github diff viewer doesn't make that clear... again, you could choose to namespace the plugin instead (or in addition to a class exists check)

cliffordp commented on readme.txt:
```diff
@@ -5,7 +5,7 @@ Tags: events, calendar
 Requires at least: 4.5
 Tested up to: 4.9.6
 Requires PHP: 5.4
```
```suggestion
Requires PHP: 5.6
```

cliffordp commented on readme.txt:
```diff
@@ -5,7 +5,7 @@ Tags: events, calendar
 Requires at least: 4.5
 Tested up to: 4.9.6
```
update to latest WP you tested with

cliffordp commented:
@andrasguseo this repo already has a `.pot` file but you'll want to regenerate it since some things in the code changed

Commit:
Update readme.txt

Co-Authored-By: Clifford <tko@tourkick.com>

Commit:
Update readme.txt

Co-Authored-By: Clifford <tko@tourkick.com>

Commit:
Update readme.txt

Commit:
Reworking the class for autoloader

Commit:
Revert "Reworking the class for autoloader"

This reverts commit 0a1c7c4a0b281e9d3b904ec39020ed084dbd5494.

Commit:
Starting to set up class and autoloader

Commit:
Define constants

Commit:
Change class name

Commit:
Adding / adjusting class check

The only thing changed is the class_exists part at the top. The rest was automatically reformatted by PHPStorm, so there is no other code change.

Commit:
Updating settings helper

Copied new settings helper and setting file. Settings will be adjusted after this.
Deleted old settings helper.

Commit:
Add $class_loader and $settings

Commit:
Changing text domain

Commit:
Added PHP version check and call to class_loader and get_settings

Commit:
Revert "Added PHP version check and call to class_loader and get_settings"

This reverts commit 1d347c87ce6dcae642237201d613e5892eb085c5.

Commit:
Added PHP version check

Commit:
Added class loader

Commit:
Added get_settings

Commit:
Adjusting options / settings handling

Commit:
Changing arrays to short syntax

Commit:
Moving settings to separate file

Commit:
Removed duplicate get_settings

Commit:
Making settings work

Commit:
Housekeeping

Commit:
Housekeeping #2

Commit:
Updated .pot and removed .po

Commit:
Text adjustment and pot update.

Commit:
Readme update

Commit:
Delete .gitignore

Commit:
Rename plugin file for consistency

Commit:
Added compatibility clarification in the description

cliffordp commented on readme.txt:
```diff
@@ -35,6 +37,10 @@ We're always interested in your feedback and our [premium forums](https://theeve
 
 == Changelog ==
 
+= 1.0.1 2020-02-14 =
```
date

cliffordp commented on src/Settings.php:
```diff
@@ -0,0 +1,317 @@
+				'start_time'  => [
+					'type'            => 'dropdown',
+					'options'         => $start_hours,
+					'label'           => esc_html__( 'Start hour', 'PLUGIN_TEXT_DOMAIN' ),
```
constant so no quotes, here and elsewhere, then regenerate .pot

cliffordp commented on src/Settings.php:
```diff
@@ -0,0 +1,317 @@
+		 *
+		 * @var string
+		 */
+		private $options_prefix = 'tribe_ext_lwh_opts_';
```
`lwh`??? should it be `lwv`, `lwvtr`, or something else?

cliffordp commented on tribe-ext-limit-week-view-time-range.php:
```diff
@@ -0,0 +1,207 @@
+namespace Tribe\Extensions\Limit_Week_View_Time_Range;
+
+use Tribe__Autoloader;
+use Tribe__Dependency;
```
Remove this line, since we don't _use_ it.

andrasguseo replied:
All 4 lines?
Or only line 19?

cliffordp replied:
Just `use Tribe__Dependency;`

In PhpStorm, it should be grayed out because you're not using it.

andrasguseo replied:
Oh wow, I need to check that feature. :) Thanks!

cliffordp commented on tribe-ext-limit-week-view-time-range.php:
```diff
@@ -0,0 +1,207 @@
+			if ( ! $this->php_version_check() ) {
+				return;
+			}
+
```
Let's add a `tribe_events_views_v2_is_enabled()` check (with `function_exists()`) and if _true_, `return;` so the rest of the plugin does not load, but then also show a notice (or not if you feel it best not to).

andrasguseo replied:
Did an "elaborate scheme" for this. :) Maybe could go into the template. Did a PR for it.

cliffordp commented on tribe-ext-limit-week-view-time-range.php:
```diff
@@ -0,0 +1,207 @@
+			$end_of_day   = $options['end_time'];
+
+			// Fallback
+			if ( ! is_numeric( $start_of_day ) || $start_of_day < 0 || $start_of_day > 23 ) {
```
one conditional per new line

cliffordp commented on tribe-ext-limit-week-view-time-range.php:
```diff
@@ -0,0 +1,207 @@
+			if ( ! is_numeric( $start_of_day ) || $start_of_day < 0 || $start_of_day > 23 ) {
+				$start_of_day = 0;
+			}
+			if ( ! is_numeric( $end_of_day ) || $end_of_day < 1 || $end_of_day > 23 || $end_of_day < $start_of_day ) {
```
one conditional per new line

cliffordp commented on tribe-ext-limit-week-view-time-range.php:
```diff
@@ -0,0 +1,207 @@
+		 *
+		 * @param $hours
+		 *
+		 * @return mixed
```
always an array, right?

cliffordp commented on tribe-ext-limit-week-view-time-range.php:
```diff
@@ -0,0 +1,207 @@
+		 *
+		 * @return mixed
+		 */
+		public function filter_week_hours( $hours ) {
```
```suggestion
		public function filter_week_hours( array $hours ) {
```

cliffordp commented on tribe-ext-limit-week-view-time-range.php:
```diff
@@ -0,0 +1,207 @@
+		}
+
+		/**
+		 * Filters the hours
```
Add trailing period.

cliffordp commented on tribe-ext-limit-week-view-time-range.php:
```diff
@@ -0,0 +1,207 @@
+		/**
+		 * Filters the hours
+		 *
+		 * @param $hours
```
```suggestion
		 * @param array $hours Add description or not, your choice. Maybe add a `@see` docblock.
```

cliffordp commented on tribe-ext-limit-week-view-time-range.php:
```diff
@@ -0,0 +1,207 @@
+		 *
+		 * @link https://secure.php.net/manual/en/migration56.new-features.php
+		 * 5.6: Variadic Functions, Argument Unpacking, and Constant Expressions
+		 *
```
Remove these lines since 5.6 is the minimum anyway.

cliffordp reviewed (dismissed):
Looking good. Keep it up.

Commit:
Updated version number and changelog to better reflect changes

Commit:
Updated version number and changelog to better reflect changes

Commit:
Removing quotes from constant

Commit:
Made plugin options prefix more clear.

Commit:
Removed unused Tribe_Dependency

Commit:
Splitting conditionals to separate lines for better readability.

Commit:
Specifying that the $hours variable is an array indeed

Commit:
Housekeeping - removing PHP notes

Commit:
Added option and method to check for view setting

Commit:
Updated pot file

juanfra commented on tribe-ext-limit-week-view-time-range.php:
```diff
@@ -0,0 +1,266 @@
+		 * @return string
+		 *
+		 */
+		public $view_needed = "V1 legacy";
```
I'd keep this to `V1` or `v1`, and single quotations mark.

juanfra commented on tribe-ext-limit-week-view-time-range.php:
```diff
@@ -0,0 +1,266 @@
+
+			$show_warning = (
+				$view_required_version == 'V1 legacy' && tribe_events_views_v2_is_enabled()
+				|| $view_required_version == 'V2 updated' && ! tribe_events_views_v2_is_enabled()
```
`$view_required_version` comes from `$this->view_needed`, which is set to `v1`, why are we checking if its value is v2? I don't see us changing this value, maybe I'm missing something?

juanfra commented on tribe-ext-limit-week-view-time-range.php:
```diff
@@ -0,0 +1,266 @@
+				if ( is_admin() && current_user_can( 'activate_plugins' ) && $show_warning ) {
+					$message = '<p>';
+					$message .= sprintf(
+						__(
+							'%s requires %s views to work. It will not work with %s views.', PLUGIN_TEXT_DOMAIN
+						),
```
```suggestion
						__( '%s requires %s views to work. It will not work with %s views.', PLUGIN_TEXT_DOMAIN ),
```

juanfra reviewed (commented):
Looking good! A few questions and suggestions.

cliffordp commented:
The comments @juanfra left were some of the same things I fixed in @andrasguseo's Extensions Template PR, awaiting another's code review (hint, hint ;)

https://github.com/mt-support/tribe-ext-extension-template/pull/18#issuecomment-613549930

Once that PR's approved and merged in, let's update this repo's code accordingly (the Views Version checking part, I mean)

andrasguseo commented:
@cliffordp can you teach me how to do that update?

cliffordp commented:
@andrasguseo implement these bits:
* https://github.com/mt-support/tribe-ext-extension-template/blob/master/tribe-ext-extension-template.php#L146
* https://github.com/mt-support/tribe-ext-extension-template/blob/master/tribe-ext-extension-template.php#L218

Commit:
Updated view check method...

... and added a check for empty $options to avoid error.

andrasguseo commented:
@cliffordp when you get a chance could you give this a quick glance?
Note, this is for V1 only.

cliffordp commented on readme.txt:
```diff
@@ -35,6 +37,10 @@ We're always interested in your feedback and our [premium forums](https://theeve
 
 == Changelog ==
 
+= 1.1.0 TBD =
```
date

cliffordp commented on readme.txt:
```diff
@@ -3,9 +3,9 @@ Contributors: ModernTribe
-Tested up to: 4.9.6
-Requires PHP: 5.4
-Stable tag: 1.0.0
+Tested up to: 5.3.2
```
version

cliffordp commented on src/Settings_Helper.php:
```diff
@@ -0,0 +1,170 @@
+			}
+
+		}
+
```
remove extra line

cliffordp commented on src/Settings_Helper.php:
```diff
@@ -0,0 +1,170 @@
+		public function remove_field( $field_key, $setting_tab ) {
+			$this->remove_fields[ $setting_tab ][] = $field_key;
+		}
+
```
remove extra line

cliffordp commented on src/Settings_Helper.php:
```diff
@@ -0,0 +1,170 @@
+
+			// Fields inserted above a neighboring field.
+			if ( array_key_exists( $tab, $this->insert_fields_above ) ) {
+
```
remove extra line

cliffordp commented on src/Settings_Helper.php:
```diff
@@ -0,0 +1,170 @@
+
+			// Fields inserted below a neighboring field.
+			if ( array_key_exists( $tab, $this->insert_fields_below ) ) {
+
```
remove extra line

cliffordp commented on src/Settings_Helper.php:
```diff
@@ -0,0 +1,170 @@
+
+			// Fields appended to misc section.
+			if ( array_key_exists( $tab, $this->insert_fields_misc ) ) {
+
```
remove extra line

cliffordp commented on src/Settings_Helper.php:
```diff
@@ -0,0 +1,170 @@
+		 * @return array $fields The fields within tribe settings page
+		 */
+		public function filter_options( $fields, $tab ) {
+
```
remove extra line

cliffordp commented on src/Settings_Helper.php:
```diff
@@ -0,0 +1,170 @@
+
+			// Fields that will be removed.
+			if ( array_key_exists( $tab, $this->remove_fields ) ) {
+
```
remove extra line

cliffordp commented on tribe-ext-limit-week-view-time-range.php:
```diff
@@ -0,0 +1,305 @@
+	class_exists( 'Tribe__Extension' )
+	&& ! class_exists( Main::class )
+) {
+
```
remove extra line

cliffordp commented on tribe-ext-limit-week-view-time-range.php:
```diff
@@ -0,0 +1,305 @@
+		 * 'V2 updated'
+		 *
+		 * @return string
+		 *
```
remove extra line

cliffordp commented on tribe-ext-limit-week-view-time-range.php:
```diff
@@ -0,0 +1,305 @@
+		private $settings;
+
+		/**
+		 * Set up if it only works with a specific view
+		 * Possible values
```
Period at end of first line. New line before additional comments.

cliffordp reviewed (changes requested):
minor tweaks requested

andrasguseo commented:
Made it into a draft.
https://github.com/mt-support/tribe-ext-limit-week-view-time-range/pull/3 will replace this.
